dll/win32/shlwapi/ordinal.c

```diff
@@ -2732,12 +2754,41 @@ DWORD WINAPI SHGetRestriction(LPCWSTR lpSubKey, LPCWSTR lpSubName, LPCWSTR lpVal
  *  different POLICYDATA structure and implements a similar algorithm adapted to
  *  that structure.
  */
+#ifdef __REACTOS__
+DWORD WINAPI
+SHRestrictionLookup(
+    _In_ DWORD policy,
+    _In_ LPCWSTR key,
+    _In_ const POLICYDATA *polTable,
+    _Inout_ LPDWORD polArr)
+#else
 DWORD WINAPI SHRestrictionLookup(
 	DWORD policy,
 	LPCWSTR initial,
 	LPPOLICYDATA polTable,
 	LPDWORD polArr)
+#endif
 {
+#ifdef __REACTOS__
+    size_t iItem = 0;
+    const POLICYDATA *entry;
+
+    TRACE("(0x%08lX, %s, %p, %p)\n", policy, debugstr_w(key), polTable, polArr);
+
+    if (!polTable->appstr)
+        return 0;
+
+    for (entry = polTable; entry->appstr != NULL; ++entry, ++iItem)
+    {
+        if (policy == entry->policy)
+        {
+            DWORD ret = polArr[iItem];
+            if (ret == SHELL_NO_POLICY)
+                ret = polArr[iItem] = SHGetRestriction(key, entry->appstr, entry->keystr);
+            return ret;
+        }
+    }
+#else
```

Comment:
Not sure why this function had to be reimplemented. The only actual change needed in the wine code would be to remove the `if (!polTable || !polArr)` check, and replace the `polTable->policy` test in the for-loop by a test against appstr.
(And I'm not sure that this last change in thee for-loop is justified as well.)

Reply:
Wine code doesn't check `entry->appstr`.

Reply:
Fixed.